File: go/functions/map_operations.go
package main

import "fmt"

func main() {
	person := map[string]interface{}{
		"name": "John",
		"age":  30,
		"city": "New York",
	}

	fmt.Printf("Person: %v\n", person)
	fmt.Printf("Name: %v\n", person["name"])
	fmt.Printf("Age: %v\n", person["age"])

	person["email"] = "john@example.com"
	fmt.Printf("After adding email: %v\n", person)

	if name, exists := person["name"]; exists {
		fmt.Printf("Name exists: %v\n", name)
	}

	fmt.Println("\nIterating over map:")
	for key, value := range person {
		fmt.Printf("  %s: %v\n", key, value)
	}

	delete(person, "email")
	fmt.Printf("After deleting email: %v\n", person)
}

File: kotlin/functions/map_operations.kt
fun main() {
    val person = mutableMapOf(
        "name" to "John",
        "age" to "30",
        "city" to "New York"
    )

    println("Person: $person")
    println("Name: ${person["name"]}")
    println("Age: ${person["age"]}")
    println("Keys: ${person.keys}")
    println("Values: ${person.values}")

    person["email"] = "john@example.com"
    println("\nAfter adding email: $person")

    if ("name" in person) {
        println("\nName exists: ${person["name"]}")
    }

    println("\nIterating over map:")
    for ((key, value) in person) {
        println("  $key: $value")
    }

    person.remove("email")
    println("\nAfter removing email: $person")

    println("Size: ${person.size}")
}
